### ISSUE TITLE
application/json charset not required

### ISSUE THREAD

**Issue by pine3ree** [slimphp/Slim#2008]
Hello,

``` php
class Response 
{
//...
$responseWithJson = $this->withHeader('Content-Type', 'application/json;charset=utf-8');
//...
```

According to the specs https://tools.ietf.org/html/rfc4627 media type application/json does not include a charset parameter as it expects json to be always UTF-(8|16|32)

It seems though that some servers/services do not work if you omit it, while other do not work if you include it.

**Comment by akrabat** [comment]
From [https://tools.ietf.org/html/rfc7159](https://tools.ietf.org/html/rfc7159#page-11):

> Note:  No "charset" parameter is defined for this registration.
>       Adding one really has no effect on compliant recipients.

I assume we should probably remove this, but I'm concerned about the BC implications if some services do not work if you omit it?

**Comment by pine3ree** [comment]
I'd say that leaving it should do no harm, since the extra parameter should be just discarded by the endpoint request handler. I stumbled upon some reported issues like [this one](https://support.zendesk.com/hc/en-us/community/posts/205218707-Zendesk-API-does-not-accept-charset-utf-8-as-part-of-Content-Type) but related to a client request.

On the other side including it could be misleading since one could think that he/she can control the charset of the response.

I was just wondering if the charset parameter was added by design in slim for a specific reason (well, many other frameworks include it as well).

kind regards
